sym_object.py

```diff
@@ -583,22 +577,22 @@ def deleteButtonPressed(self, event):
         # true
         if (click_x > delete_button_x and click_x < delete_button_x + \
             delete_button_width and click_y > delete_button_y and click_y < \
-            delete_button_y + delete_button_width):
+            delete_button_y + delete_button_width) and self in self.state.selected_sym_objects:
             return True
 
         return False
 
     def doesOverlap(self, item):
         """checks if two objects overlap"""
 
-        l1_x = self.scenePos().x()
-        l1_y = self.scenePos().y()
-        r1_x = self.scenePos().x() + self.width
-        r1_y = self.scenePos().y() + self.height
-        l2_x = item.scenePos().x()
-        l2_y = item.scenePos().y()
-        r2_x = item.scenePos().x() + item.width
-        r2_y = item.scenePos().y() + item.height
+        l1_x = self.mapToScene(self.rect.rect()).boundingRect().left()
+        l1_y = self.mapToScene(self.rect.rect()).boundingRect().top()
+        r1_x = self.mapToScene(self.rect.rect()).boundingRect().right()
+        r1_y = self.mapToScene(self.rect.rect()).boundingRect().bottom()
+        l2_x = item.mapToScene(item.rect.rect()).boundingRect().left()
+        l2_y = item.mapToScene(item.rect.rect()).boundingRect().top()
+        r2_x = item.mapToScene(item.rect.rect()).boundingRect().right()
+        r2_y = item.mapToScene(item.rect.rect()).boundingRect().bottom()
```

Comment:
put this in its own function? just a suggestion

Reply:
good thinking